client/main.go

```diff
@@ -59,21 +60,21 @@ type UserHandleFunc func(*bufio.Writer, string)
 func handleServerInput(reader *bufio.Reader, handlers map[scs.ActionName]ServerHandleFunc, state *State) {
 	for {
 		action := &scs.Action{}
-		err := gob.NewDecoder(reader).Decode(action)
-		switch {
-		case err == io.EOF:
-			log.Println("Reached EOF - close this connection.\n   ---")
-			return
-		case err != nil:
-			log.Println("\nError reading command.", err)
+		if err := gob.NewDecoder(reader).Decode(action); err != nil {
+			if err == io.EOF {
+				log.Println("Reached EOF - close this connection.\n   ---")
+			} else {
+				log.Println("\nError reading command.", err)
+			}
 			return
 		}
+
 		handlerFunc := handlers[action.Name]
-		if handlerFunc != nil {
-			handlerFunc(reader, state)
-		} else {
+		if handlerFunc == nil {
```

Comment:
Another option is to use the second value from getting from a map. More useful when you can't distinguish between no value and the [zero value](https://golang.org/ref/spec#The_zero_value), like `map[string]int`.

```go
handlerFunc, ok := handlers[action.Name]
if !ok {
	log.Println("Unidentified command")
	continue
}
```